client-sdk/go/modules/core/types.go

```diff
@@ -12,3 +13,29 @@ type EstimateGasQuery struct {
 	// Tx is the unsigned transaction to estimate.
 	Tx *types.Transaction `json:"tx"`
 }
+
+type RuntimeInfoResponse struct {
+	RuntimeVersion *version.Version      `json:"runtime_version"`
+	StateVersion   uint32                `json:"state_version"`
+	Modules        map[string]ModuleInfo `json:"modules"`
+}
+
+type ModuleInfo struct {
+	Version uint32              `json:"version"`
+	Params  []uint8             `json:"params"`
+	Methods []MethodHandlerInfo `json:"methods"`
+}
+
+type MethodHandlerInfo struct {
+	Name string            `json:"name"`
+	Kind methodHandlerKind `json:"kind"`
```

Comment:
Public struct has a field with a private type? That's weird.

Reply:
I saw that online as a suggestion for emulating enums. The idea is that you can never create a `methodHandlerKind` from scratch, you need to use one of the predefined constants, which serve as enum values. It seems to mesh well with our cbor enums.
I'm also open to other approaches tho.